adoc/chapters/programming_interface.adoc

```diff
@@ -716,6 +716,22 @@ Construct a SYCL [code]#property_list# with zero or more properties.
 
 |====
 
+=== Information queries
+
+Several classes in SYCL provide a generic mechanism for querying the class for
+information.
+
+Each available query is described by an <<information-descriptor>>, which is a
+class or class template that encapsulates an information query and its return
+type.
+
+==== Information query interface
+
+The information query interface consists of a [code]#get_info()# function
+template, which is templated on an <<information-descriptor>>.
```

Comment:
This isn't entirely true because there is also `get_backend_info`, which also takes an information descriptor.

Reply:
Good catch!  I always forget about this backend-specific stuff.  I've added references to backends in 2aa55a09faf23e1319573ec4a3b14cdc112d54bd.